homeassistant/components/dialogflow/__init__.py

```diff
@@ -153,16 +169,43 @@ def __init__(self, parameters, api_version):
     def add_speech(self, text):
         """Add speech to the response."""
         assert self.speech is None
-
         if isinstance(text, template.Template):
             text = text.async_render(self.parameters, parse_result=False)
-
         self.speech = text
+        
+    def add_next_scene(self, next_scene):
+        """Add next_scene to the response."""
+        self.next_scene = next_scene
 
     def as_dict(self):
         """Return response in a Dialogflow valid dictionary."""
+        return_value = None
         if self.api_version is V1:
-            return {"speech": self.speech, "displayText": self.speech, "source": SOURCE}
+            return_value = {"speech": self.speech, "displayText": self.speech, "source": SOURCE}
 
         if self.api_version is V2:
-            return {"fulfillmentText": self.speech, "source": SOURCE}
+            return_value = {"fulfillmentText": self.speech, "source": SOURCE}
+
+        if self.api_version is V3:
+            return_value = {
+                "session": {
+                "id": self.parameters["session_id"],
+                "params": {}
+                },
+                "prompt": {
+                "override": False,
+                "firstSimple": {
+                  "speech": self.speech,
+                  "text": ""
+                }
+                },
+                "scene": {
+                "name": "SceneName",
+                "slots": {},
+                "next": {
+                  "name": self.next_scene
+                }
+                }
+            }
+        _LOGGER.warning( "valRetour : %s" %( return_value ))
```

Comment:
You are still logging as a warning here. Also you ignored my comment on early return like the original code did.

Reply:
return change, and warning delete